src/audio/Kconfig

```diff
@@ -75,4 +75,134 @@ config COMP_TEST_KEYPHRASE
 	  Select for KEYPHRASE_TEST component.
 	  Provides basic functionality for use in testing of keyphrase detection pipelines.
 
-endmenu
+config COMP_ASRC
+	bool "ASRC component"
+	default y
+	help
+	  Select for Asynchronous sample rate conversion (ASRC)
+	  component. The ASRC has the capabilities of (synchronous)
+	  SRC but can in addition track a slave DAI that is not in
+	  sync with firmware timer. The ASRC can be setup for
+	  synchronous operation without DAI tracking via topology
+	  though it has a bit higher computational load than SRC. The
+	  RAM footprint of ASRC is smaller than SRC because it does
+	  not have pre-computed filter coefficients for every
+	  conversion fraction that SRC does.
+
+if COMP_ASRC
+
+choice
+        prompt "ASRC down sampling conversions set"
+        default COMP_ASRC_DOWNSAMPLING_FULL
+
+config COMP_ASRC_DOWNSAMPLING_FULL
+	bool "Full downsampling conversions support"
+	select ASRC_SUPPORT_CONVERSION_24000_TO_08000
+	select ASRC_SUPPORT_CONVERSION_24000_TO_16000
+	select ASRC_SUPPORT_CONVERSION_48000_TO_08000
+	select ASRC_SUPPORT_CONVERSION_48000_TO_11025
+	select ASRC_SUPPORT_CONVERSION_48000_TO_12000
+	select ASRC_SUPPORT_CONVERSION_48000_TO_16000
+	select ASRC_SUPPORT_CONVERSION_48000_TO_22050
+	select ASRC_SUPPORT_CONVERSION_48000_TO_24000
+	select ASRC_SUPPORT_CONVERSION_48000_TO_32000
+	select ASRC_SUPPORT_CONVERSION_48000_TO_44100
+	help
+	  This option enables all supported downsampling conversions
+	  for the asynchronous sample rate conversion. All the
+	  upsampling conversions use the same filter always enabled
+	  coefficients set so there is no configuration options for
+	  them.
+
+config COMP_ASRC_DOWNSAMPLING_CUSTOM
+	bool "Custom downsampling conversions support"
```

Comment:
depends on COMP_ASRC_DOWNSAMPLING_FULL=n?

Reply:
I tried to add this but it creates a dependency loop. Can you help further how to accomplish that. Here's what I tried:

```
diff --git a/src/audio/Kconfig b/src/audio/Kconfig
index eaaf10c8321a..8674e4997e77 100644
--- a/src/audio/Kconfig
+++ b/src/audio/Kconfig
@@ -116,6 +116,7 @@ config COMP_ASRC_DOWNSAMPLING_FULL
 
 config COMP_ASRC_DOWNSAMPLING_CUSTOM
        bool "Custom downsampling conversions support"
+       depends on COMP_ASRC_DOWNSAMPLING_FULL=n
        help
          This option used to manually select each downsampling ratio.
          In order to optimize the text code size of the sample rate
```

Reply:
Why don't you just make them `choice` like in dmic configs?

Reply:
Umm, we already have two choises:  COMP_ASRC_DOWNSAMPLING_FULL, and COMP_ASRC_DOWNSAMPLING_CUSTOM.  To me the ASRC menus structure looks identical to DMIC. I don't know what improvement @plbossart 's suggestion would do for menuconfig usability. I'm kconfig noob so can you please explain!